### ISSUE TITLE
replace job_tracker with job_tracker_v2

### ISSUE THREAD

**Issue by bossie** [Open-EO/openeo-geopyspark-driver#387]
I have a couple of issues that directly or indirectly rely on `job_tracker_v2` e.g. #325 and #281.

Also @JanssenBrm had trouble finding out why his batch jobs kept failing but this was fixed in #292 for `job_tracker_v2`:
> I am having issues with executing my batch jobs. They seem to be failing without a clear error in the logs (example: j-e384ff99e7a841eb9a598abe5131635e)

Time to put the original one to rest.

**Comment by bossie** [comment]
Simply removing the original `job_tracker` means that ETL API usage reporting won't work anymore (on YARN) so #325 has to be merged first.

ETL API usage reporting was never done on CreoDIAS anyway so it should be safe to replace it there.

**Comment by bossie** [comment]
For reference: adding the job_tracker to CreoDIAS as an external process was done in #235.

**Comment by bossie** [comment]
`job_tracker_v2` is operational on CreoDIAS:
![job_tracker_v2_creo](https://user-images.githubusercontent.com/1032518/230084570-42d1bb95-3af2-49e4-8f44-ec722e6ffd9d.png)

**Comment by bossie** [comment]
The fix implemented now puts the burden of looking up the ETL API credentials on Terraform. This might have to be improved upon.

I noticed that the EJR suffers from the same problem; it manifests itself in a different way, however:
```
Traceback (most recent call last):
  File "/opt/openeo/lib/python3.8/site-packages/openeo_driver/util/logging.py", line 327, in just_log_exceptions
    yield
  File "/opt/openeo/lib/python3.8/site-packages/openeogeotrellis/backend.py", line 1006, in get_elastic_job_registry
    ejr_creds = vault.get_elastic_job_registry_credentials()
  File "/opt/openeo/lib/python3.8/site-packages/openeogeotrellis/vault.py", line 97, in get_elastic_job_registry_credentials
    client = self._client(token=vault_token or self.login_kerberos())
  File "/opt/openeo/lib/python3.8/site-packages/openeogeotrellis/vault.py", line 81, in login_kerberos
    vault_token = subprocess.check_output(cmd, text=True, stderr=PIPE)
  File "/usr/lib64/python3.8/subprocess.py", line 415, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib64/python3.8/subprocess.py", line 493, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/usr/lib64/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib64/python3.8/subprocess.py", line 1706, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'vault'
```

**Comment by bossie** [comment]
A problem with the kubecost API prevents the `job_tracker_v2` from making any progress on CreoDIAS:

```
Traceback (most recent call last):
  File "/opt/openeo/lib/python3.8/site-packages/openeogeotrellis/job_tracker_v2.py", line 298, in _get_usage
    raise K8sException(
K8sException: Unexpected response {'code': 200, 'data': [{}]}
```

This leads to a `_Usage` of `None`, preventing a status update further on:
```
Traceback (most recent call last):
  File "/opt/openeo/lib/python3.8/site-packages/openeogeotrellis/job_tracker_v2.py", line 388, in update_statuses
    self._sync_job_status(
  File "/opt/openeo/lib/python3.8/site-packages/openeogeotrellis/job_tracker_v2.py", line 516, in _sync_job_status
    usage=job_metadata.usage.to_dict(),
AttributeError: 'NoneType' object has no attribute 'to_dict'
```

`job_metadata.usage` can be `None` so that's a bug.

**Comment by bossie** [comment]
After a few fixes, including those in #390, both Terrascope and CreoDIAS now run `job_tracker_v2`.
